modules/baudiPersons.xqm

```diff
@@ -237,10 +236,10 @@ declare function baudiPersons:getOccupation($persId as xs:string) {
 declare function baudiPersons:getResidences($persId as xs:string) {
     let $person := $app:collectionPersons/id($persId)
     return
-        if(if($person//tei:recidence[. != '']) then(true()) else(false()))
-        then(<ul>{for $recidence in $person//tei:recidence[. != '']
+        if(exists($person//tei:residence[. != '']))
+        then(<ul>{for $residence in $person//tei:residence[. != '']
                                 return
-                                    <li class="baudiListItem">{$recidence/text()}</li>
+                                    <li class="baudiListItem">{$residence/text()}</li>
                               }</ul>)
         else()
 };
```

Comment:
@dpleipold Thank you for the necessary adjustments! Another thing: Just a question (not tested yet): Do you expect this to work? [Skipping the conditional by using the where-"filter"]
```
declare function baudiPersons:getResidences($persId as xs:string) {
    let $person := $app:collectionPersons/id($persId)
    where exists($person//tei:residence[. != ''])
    return
         <ul>{for $residence in $person//tei:residence[. != '']
                      return
                          <li class="baudiListItem">{$residence/text()
                }</li>
         </ul>
};
```

Just a spontanious idea to get rid of the conditional. I'm going to test it soon with the whole PR!

Reply:
just for protocol, yes it works.